app/services/dialog_local_service.rb

```diff
@@ -78,7 +40,7 @@ def determine_dialog_locals_for_custom_button(obj, button_name, resource_action,
   private
 
   def determine_api_endpoints(obj, display_options = {})
-    base_name = obj.class.name.demodulize
+    base_name = obj.class.base_class.name
```

Comment:
Does `base_model` work better here?  I know it will separate `Vm` and `MiqTemplates` models instead of having to filter on `VmOrTemplate`.

But it will not help for `ExtManagementSystem`:
```ruby
irb(main):026:0> ManageIQ::Providers::CloudManager.base_model.name
=> "ExtManagementSystem"

irb(main):027:0> ManageIQ::Providers::InfraManager.base_model.name
=> "ExtManagementSystem"
```